### PR TITLE
Fetch exchange rates in bulk from synth

### LINKED ISSUE #1047: Make bulk request for exchange rates to Synth API
**Describe the bug**
See title.

**To Reproduce**
1. Set up project without synth API key first
2. Try to convert a currency on the App and get the expected error message
3. Then adding API key + restart server
4. Open the web app
5. Noticed the 700+ API calls

**Screenshots / Recordings**
image

### PR DESCRIPTION
**Pull request by tonyvince**
Fix #1047

### CODE REVIEW

File: app/models/provider/synth.rb [hunk 1]
```diff
@@ -57,12 +57,40 @@ def fetch_exchange_rate(from:, to:, date:)
     end
   end
 
+  def fetch_exchange_rate_for_date_range(from:, to:, date_start:, date_end:)
```
Comment (zachgoll) [inline app/models/provider/synth.rb:60]: Mostly style here, but could we keep this consistent with the `fetch_security_prices` method stub?
```suggestion
  def fetch_exchange_rates(from:, to:, start_date:, end_date:)
```

File: app/models/exchange_rate/provided.rb [hunk 2]
```diff
@@ -6,12 +6,31 @@ module ExchangeRate::Provided
...
+            rate = ExchangeRate.new \
+              from_currency: from,
+              to_currency: to,
+              date: exchange_rate.dig(:date).to_date,
+              rate: exchange_rate.dig(:rate)
+
+            rate.save! if cache
+            rate
+          rescue ActiveRecord::RecordNotUnique
+            next
```
Comment (zachgoll) [inline app/models/exchange_rate/provided.rb:29]: I think this should fix #1037 as well correct?

**Review by zachgoll — COMMENTED** [review]
Awesome work here!

File: app/models/exchange_rate/provided.rb (hunk 2 above)
Comment (tonyvince) [reply app/models/exchange_rate/provided.rb:29]: Yes!

File: app/models/provider/synth.rb (hunk 1 above)
Comment (tonyvince) [reply app/models/provider/synth.rb]: Done.

**Pull request merged**